org.lflang/src/org/lflang/generator/c/CGenerator.java

```diff
@@ -1086,7 +1086,14 @@ private void pickCompilePlatform() {
         if (targetConfig.platform != Platform.AUTO) {
             osName = targetConfig.platform.toString();
         }
-        if (osName.contains("mac") || osName.contains("darwin")) {
+        
+        if (osName.contains("arduino")) {
+            if (mainDef != null && !targetConfig.useCmake) {
```

Comment:
Does the Arduino support work with `cmake: false` (i.e., without CMake)? If not, there is no need for this.

Reply:
Taken out since we only support CMake for the time being